src/app/features/tech-record/components/vehicle-technical-record/vehicle-technical-record.component.ts

```diff
@@ -166,4 +171,28 @@ export class VehicleTechnicalRecordComponent implements OnInit, OnDestroy {
       : 'This vehicle does not have enough information to be tested.'
           + ' Call the Contact Centre to complete this record so tests can be recorded against it.';
   }
+
+  isADRVehicleType(): boolean {
+    return this.techRecord?.techRecord_vehicleType === 'hgv'
+    || this.techRecord?.techRecord_vehicleType === 'lgv'
+    || this.techRecord?.techRecord_vehicleType === 'trl';
+  }
+
+  showGenerateADRCertificateButton(): boolean {
+    const isNotArchivedAndEditing = !this.isArchived && !this.isEditing;
+    return this.isADREnabled && this.isADRVehicleType() && isNotArchivedAndEditing;
+  }
+
+  validateADRDetails(): void {
```

Comment:
validateADRDetailsAndNavigate may be a more intuitive function name for this, just to encapsulate the navigation behaviour